narwhal/consensus/src/tests/randomized_tests.rs

```diff
@@ -403,7 +404,11 @@ pub fn make_certificates_with_parameters(
                 parents_digests.insert(0, my_parent_digest);
             }
 
-            assert!(parents_digests.len() >= committee.quorum_threshold() as usize);
+            assert!(
+                parents_digests.len() >= committee.quorum_threshold() as usize,
+                "Failed on seed {}. At least 2f+1 parents are needed.",
+                seed
```

Comment:
Would this potentially cause flakiness where this sometimes fails? Can we instead re-generate the inputs if that is the case that some seeds fail?

Reply:
As we discussed offline the algorithm should always produce `2f+1` parents - if it doesn't then this is actually an issue/bug and this assert here ensures that we catch that.